### PR TITLE
Doc/improve wfspec docs

### PR DESCRIPTION
**Pull request by coltmcnealy-lh**

### CODE REVIEW

File: docs/docs/05-developer-guide/05-task-worker-development.md [hunk 1]
```diff
@@ -160,37 +144,108 @@ As described in our [Failure Handling Concept Docs](/docs/concepts/exception-han
...
+
+    return f"successfully shipped {item_sku}!"
 ```
 
   </TabItem>
 </Tabs>
 
+The first argument to the `LHTaskException` constructor is the _name_ of the `EXCEPTION` we are going to throw. This is useful if you want to be able to catch specific `EXCEPTION`s with specific types in your [Failure Handlers](./08-wfspec-development/06-exception-handling.md). The second argument is a _human-readable_ error message that shows up on the `NodeRun`'s output as the `error_message` field, which is useful for debugging purposes.
+
+If you want to throw a `Failure` that has content which can be caught in your Failure Handler using the `INPUT` variable name, you can use a _third_ argument named `content`. It is optional in python and is available in an overloaded method signature in Java. The below is an example of how you might throw such an `EXCEPTION`:
```
Comment (diablouma) [inline docs/docs/05-developer-guide/05-task-worker-development.md:196]: Would be cool to point the users to the example where we show how to use that content?

File: docs/docs/05-developer-guide/08-wfspec-development/01-basics.md [hunk 2]
```diff
@@ -9,6 +12,118 @@ The `Workflow` object translates your `ThreadFunc` into a `WfSpec`. As per the [
...
+        LHConfig config = new LHConfig();
+
+        // The `Workflow` object uses the DSL to compile the WfSpec
+        Workflow workflowGenerator = Workflow.newWorkflow("my-wf-spec", Main::wfLogic);
+
+        // Convenience method to register the `WfSpec` automatically.
+        workflowGenerator.registerWfSpec(config.getBlockingStub());
+    }
+
+    // NOTE: this can be static or non-static.
```
Comment (diablouma) [inline docs/docs/05-developer-guide/08-wfspec-development/01-basics.md:48]: :smile:

File: docs/docs/05-developer-guide/08-wfspec-development/04-external-events.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # External Events
 
+:::info
+Before you can use an `ExternalEventDef` in a `WfSpec`, you need to create the `ExternalEventDef` metadata object in LittleHorse. You can do that following our [metadata management docs](../09-grpc/05-managing-metadata.md#externaleventdef).
+:::
```
Comment (diablouma) [inline docs/docs/05-developer-guide/08-wfspec-development/04-external-events.md:5]: :smile: thanks for adding this, is really useful

File: docs/docs/05-developer-guide/08-wfspec-development/06-exception-handling.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ The syntax to handle a `Failure` is similar no matter which type of `Node` you a
 This section is concerned with throwing an `EXCEPTION` at the `ThreadSpec` level inside a `WfSpec`. If you want to throw an `EXCEPTION` at the Task Worker level, please refer to the [Task Worker Development Docs](/docs/developer-guide/task-worker-development#throwing-workflow-exceptions)
 :::
 
-In programming languages such as Java and Python (please, let's not mention Go...I have opinions about Go but it's best I don't discuss it), you can `throw` or `raise` an `Exception`. For example:
+In well-designed programming languages such as Java and Python, you can `throw` or `raise` an `Exception`. For example:
```
Comment (diablouma) [inline docs/docs/05-developer-guide/08-wfspec-development/06-exception-handling.md:30]: I think the 'well-designed' is something we could exclude from these instructions. If we have people that really like GoLang can interpret this in a way we don't want. WDYT?

**Review by diablouma — COMMENTED** [review]
left some comments :wink:

File: docs/docs/05-developer-guide/08-wfspec-development/06-exception-handling.md (hunk 4 above)
Comment (coltmcnealy-lh) [reply docs/docs/05-developer-guide/08-wfspec-development/06-exception-handling.md:30]: :joy:

File: docs/docs/05-developer-guide/05-task-worker-development.md (hunk 1 above)
Comment (coltmcnealy-lh) [reply docs/docs/05-developer-guide/05-task-worker-development.md:196]: That example isn't ready yet...working on it.

**Review by diablouma — APPROVED** [review]

**Pull request merged**